server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/FragmentInstanceDispatcherImpl.java

```diff
@@ -118,20 +124,67 @@ private Future<FragInstanceDispatchResult> dispatchRead(List<FragmentInstance> i
     return immediateFuture(new FragInstanceDispatchResult(true));
   }
 
-  private Future<FragInstanceDispatchResult> dispatchWriteSync(List<FragmentInstance> instances) {
+  private Future<FragInstanceDispatchResult> dispatchWriteAsync(List<FragmentInstance> instances) {
+    List<Future<FragInstanceDispatchResult>> futureList = new ArrayList<>(instances.size());
     for (FragmentInstance instance : instances) {
-      try (SetThreadName threadName = new SetThreadName(instance.getId().getFullId())) {
-        dispatchOneInstance(instance);
-      } catch (FragmentInstanceDispatchException e) {
-        return immediateFuture(new FragInstanceDispatchResult(e.getFailureStatus()));
-      } catch (Throwable t) {
-        logger.warn("[DispatchFailed]", t);
-        return immediateFuture(
-            new FragInstanceDispatchResult(
-                RpcUtils.getStatus(
-                    TSStatusCode.INTERNAL_SERVER_ERROR, "Unexpected errors: " + t.getMessage())));
+      futureList.add(
+          writeOperationExecutor.submit(
```

Comment:
I suspect that such a change would result in a performance rollback in a multi-node production cluster with high concurrent writes.

The thread executing this function is `ClientRPC-Processor`, and the maximum number of `ClientRPC-Processor` threads in threadPool is 65535. However, The coreSize of `writeOperationExecutor` threadPool is `50`. 

Prior to this change, if the concurrency was high enough, the OS could schedule up to 65535 threads despite a lot of synchronous IO, so sending RPCs could still be done in parallel. However, with this change, a maximum of 50 RPCs can currently be sent in parallel, and the remaining requests are blocked, and the higher the network latency, the more obvious this problem is.

For these reasons, I recommend using Thrift `AsyncClient`. Although cumbersome, the significant performance gains on this critical path are well worth it.

Reply:
Such implementation has been reverted, the async client will be implemented in next pr.